### DISCUSSION TITLE
How would you create a release build from command prompt? (or any other way)

### DISCUSSION THREAD

**Discussion by ChanceVincent** [stride3d/stride#1558] [General]
The Stride manual indicates that creating the release build of a stride project requires running visual studio. This unfortunately means people who can't run visual studio also can't create a release build for any stride project they create, and so there is no point in using stride at all.

However, visual studio actually does nothing special, if I understand it correctly. It Only calls MSBuild.exe with the stride project's solution file and appends the build configuration you set in visual studio settings. MSBuild.exe can be run from command prompt. I think this should mean that it should be possible to create a release build **without** requiring to run visual studio. I have been trying to figure out how to do this, but I haven't had any luck finding any info online about how to use MSBuild.exe that way. It also seems that newer versions of MSBuild.exe (ie v17 from .net sdk 6.0, which the current stride requires) do not replace/update the default system version that is accessed through command prompt, so there needs to be a way to do it with the correct version instead of the default version.

Does anyone with more experience on this subject have any guidance? I am not an expert programmer and am probably punching above my weightclass on this, but I can't run visual studio, and whether it can be possible to create the release build by another means will determine whether I can make use of stride. I have already tried simply using another IDE (sharpdevelop), but it can't open the solution file and attempting to even makes the project no longer openable with stride.

**Comment by manio143** [comment]
In .NET Core and 5+ everything can be done with the `dotnet` command. To build a release version see `dotnet build --configuration Release`.

**Reply by ChanceVincent** [reply]
Okay, that's perfect. I've just read through the references for this and made some attempts and I haven't gotten it to actually produce a build yet. Or if it has, I have no idea where it is outputting to. So this is going to take me some time to figure out, but it looks like it is much closer than I've been until now. Thank you for replying so quickly and directly to the subject. I really hope this can lead to what I need.

**Reply by ChanceVincent** [reply]
I'm sorry to report that this is actually incorrect. It seems there is no way to produce a build of a Stride project from dotnet CLI. Apparently there is in fact something special about the build tool in visual studio that makes it work where `dotnet build` does not.

**Reply by manio143** [reply]
Can you be more specific regarding the differences of output produces by VS and CLI?

**Reply by ChanceVincent** [reply]
I can't run visual studio, so I don't know what the output from that might be. Using dotnet cli responds with this:
`...\NuGet.targets(132,5):warning : Unable to find a project to restore!`
I've web searched the warning message, reviewed the dotnet cli reference, and asked about it on the dotnet section of superuser, but through all of that I have come up with no solution to it.

**Reply by manio143** [reply]
That doesn't sound familiar. Can you make sure you're running the build in the directory of .Windows project of your game? And provide more details on the output. You can change the verbosity setting for dotnet build, see documentation for referee.
